README.md

```diff
@@ -13,13 +15,49 @@ This Rails project:
 - has scaffolding, standard CRUD actions and shopping cart (without purchasing part)
 - includes: products, orders, line items, users, store, log in, log out, forms, sessions, admin, unit tests, integration tests, order notifications, etc.
 
-## To run the application:
+## To run the web app that is in a Docker container and Deployed on AWS
+
+1. Create an EC2 instance on AWS with an Ubuntu image and at least a t2.micro size.
+
+2. Create a security rule for TCP to use port 48019 and use that with the instance.
+
+3. ssh to the EC2 instance, e.g.,
+   `ssh -i <full path to .pem key> ubuntu@<IPaddress>`
+
+4. Create a `.env` file on EC2 instance with environment variables similar to the example below. Assign values to the MySQL root user and the MySQL user along with **passwords**, which **should NOT be shown to anyone**.
+
+  ```
+    MYSQL_ROOT_USER=root
+    MYSQL_ROOT_PASSWORD=therootuserpassword
+    MYSQL_USER=yourmysqluser
+    MYSQL_PASSWORD=yourmysqluserpassword
+    MYSQL_DATABASE=rails_buy_stuff_development
+    PORT_ASSIGNED_ON_HOST_FOR_DB=48018
+    PORT_ASSIGNED_ON_HOST_FOR_WEB_APP=48019
+    PORT_RAILS_WEB_SERVER_DEFAULT=3000
+    PORT_MYSQL_SERVER_DEFAULT=3306
+  ```
```

Comment:
Suggestion: you can automate creating this file. All you need is a way to generate decent passwords, and you can do that with the `openssl rand` command like this:

```
openssl rand -base64 12
```

This is the second answer on this StackExchange page and it's a lot easier than using `/dev/urandom` and `tr`: https://unix.stackexchange.com/questions/230673/how-to-generate-a-random-string

In `main.sh` you could generate the file on your mac and use `scp` to securely copy it onto the EC2 instance. Then you wouldn't need to `ssh` to the instance and edit this file. You can exploit substitutions in here-documents to generate the passwords in the right places:

```
cat > ec2-env <<EOF
MYSQL_ROOT_USER=root
MYSQL_ROOT_PASSWORD=$(openssl rand -base64 12)
MYSQL_USER=yourmysqluser
MYSQL_PASSWORD=$(openssl rand -base64 12)
MYSQL_DATABASE=rails_buy_stuff_development
PORT_ASSIGNED_ON_HOST_FOR_DB=48018
PORT_ASSIGNED_ON_HOST_FOR_WEB_APP=48019
PORT_RAILS_WEB_SERVER_DEFAULT=3000
PORT_MYSQL_SERVER_DEFAULT=3306
EOF

scp ec2-env ubuntu@<IP>:.env
```

I guess you'd also need to settle on a value for the `MYSQL_USER` but you can decide on a user name like `buystuff` in advance if you like.

I used `ec2-env` for the name of the local file but you can name it whatever you want or even use `mktemp` to create a temporary file that you can then delete at the end.

Reply:
Oh, I didn't include the `ssh` identity in the `scp` command, though I should have. It takes an `-i` parameter for the identity (i.e. private key) file just like the `ssh` command does, if I remember correctly.

Reply:
That is a good idea to use generated passwords to run the app on AWS. Thanks.

Reply:
It makes things a bit more convenient, and at least you're not reusing passwords. Like we discussed in a more realistic situation you'd probably have a secure way to fetch passwords from some secrets-keeping service, but this is OK for a demonstration.

Reply:
Changes look good to me, marking this issue resolved.